### ISSUE TITLE
Deserialize anyOf or oneOf not obeying schema additional properties

### ISSUE THREAD

**Issue by DanStory** [byu-oit/openapi-enforcer#69]
It looks like deserialize is looking in the wrong place for additional properties of schema.
https://github.com/byu-oit/openapi-enforcer/blob/master/src/schema/deserialize.js#L86

Shouldn't that be `subSchema.additionalProperties === false`?

**Comment by DanStory** [comment]
Additionally, it doesn't apply required fields as part of the scoring.

With a schema of:
```
{
    "schema": {
        "oneOf": [
            {
                "properties": {
                    "one": {
                        "type": "string"
                    },
                },
                "required": [
                    "one"
                ],
                "additionalProperties": false,
                "type": "object"
            },
            {
                "properties": {
                    "one": {
                        "type": "string"
                    },
                    "two": {
                        "type": "string"
                    }
                },
                "required": [
                    "one",
                    "two"
                ],
                "additionalProperties": false,
                "type": "object"
            }
        ]
    }
}
```

and a body of:
```
{
	"one": "value"
}
```

Causes both schema models to have a score of 2, even though the body doesn't satisfy the second model requiring two properties

**Comment by Gi60s** [comment]
Wow, I can tell you really dug into the root cause of this. I'm impressed that you were able to identify this so well. I think  you're correct, but I haven't had a chance to look yet.

I'm pretty booked most of today so I'm not sure if I'll be able to resolve this today, but it is a pretty minor change so I'll see what I can do.

Thanks for reporting the issue and taking the time to identify the cause.

**Comment by Gi60s** [comment]
OK, I believe this is fixed. I've published to NPM with version `1.10.1`.

**Comment by DanStory** [comment]
You're awesome! Thanks for your time, I verified that it is now working in our use case.
Thanks for the library too :)
